profiling/src/collections/identifiable/string_table.rs

```diff
@@ -0,0 +1,266 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+use super::*;
+use bumpalo::Bump;
+use ouroboros::self_referencing;
+use std::collections::HashMap;
+use std::hash::BuildHasherDefault;
+
+#[cfg(test)]
+use std::ops::Range;
+
+struct BorrowedStringTable<'b> {
+    /// The arena to store the characters in.
+    arena: &'b Bump,
+
+    /// Used to have efficient lookup by [StringId], and to provide an
+    /// [Iterator] over the strings.
+    vec: Vec<&'b str>,
```

Comment:
No, because the vector will resize. If we have a 32 element vec and it needs to resize to 64, the 32 elements are going to stay where they are and a new chunk of 64 elements will get allocated, and now there are 96 elements in the arena to store 64 items.

Reply:
So this would make sense in a `mmap` world, but not in the current arena allocator